experimental/dds/tree2/src/simple-tree/toMapTree.ts

```diff
@@ -161,13 +164,44 @@ function valueToMapTree(
 	const schema = getSchema(context, type);
 	assert(allowsValue(schema.leafValue, value), "Unsupported schema for provided primitive.");
 
+	const mappedValue = mapUnsupportedPrimitive(value, typeSet);
+
 	return {
-		value,
+		value: mappedValue,
 		type,
 		fields: new Map(),
 	};
 }
 
+/**
+ * Checks an incoming value to ensure it is compatible with our serialization format.
+ * For unsupported values with a schema-compatible replacement, return the replacement value.
+ * For unsupported values without a schema-compatible replacement, throw.
+ * For supported values, return the input.
+ */
+function mapUnsupportedPrimitive(
+	// eslint-disable-next-line @rushstack/no-new-null
+	value: boolean | number | string | IFluidHandle | null,
+	typeSet: TreeTypeSet,
+	// eslint-disable-next-line @rushstack/no-new-null
+): boolean | number | string | IFluidHandle | null {
+	if (Object.is(value, -0)) {
+		// Our serialized data format does not support -0.
+		// Map such input to +0.
+		return 0;
+	} else if (Number.isNaN(value) || !Number.isFinite(value)) {
+		// Our serialized data format does not support NaN nor +/-∞.
```

Comment:
Did you choose this policy because it's what JSON does when round-tripping? Or something else? Without thinking about it super hard, I think I'd just expect "NaN" to always throw, even though it's sort of nullish. I think I'd definitely be suprised if my `Infinity` turned into null. That one I don't think I ever want (but again - is that what JSON does? That might justify it... maybe).

Reply:
Yes, this is emulating JSON behavior. I'll call that out specifically in a comment.

See the task I'm wortking from for specifics: https://dev.azure.com/fluidframework/internal/_workitems/edit/6217

Reply:
Sounds good!